client/network/src/protocol/generic_proto/upgrade/notifications.rs

```diff
@@ -148,8 +153,11 @@ where TSubstream: AsyncRead + AsyncWrite + Unpin + Send + 'static,
 				socket.read_exact(&mut initial_message).await?;
 			}
 
+			let mut codec = UviBytes::default();
+			codec.set_max_len(usize::try_from(self.max_notification_size).unwrap_or(usize::max_value()));
```

Comment:
Why do we need that `try_from`? Why not simply make `max_notification_size` an usize to begin with?

Reply:
The size of something that is transmitted on the network should never be a `usize`, as we want this size limit to be the same for all participants.
However a buffer length, however, is correctly a `usize`.